### ISSUE TITLE
B006 

### ISSUE THREAD

**Issue by omgMath** [astral-sh/ruff#11030]
Hi
I found a case where my expectations does not meet the results reported by ruff. But first, let's start with the issue basics:
* Keywords searched: "B006", "mutable default"
* Ruff version: `ruff 0.4.0`
* Command invoked: `ruff check case.py --select ALL --ignore D,FA --isolated`

Where my `case.py` contains the following lines:
```python
DEFAULT_ENTRIES = [1, 2, 3]

def append_one(entries: list[int] = DEFAULT_ENTRIES) -> None:
    entries.append(1)
```

I would have hoped that this would be detected by [rule B006](https://docs.astral.sh/ruff/rules/mutable-argument-default/) (or any other rule), but sadly it is not.

However, the rule detects the issue if instead of `DEFAULT_ENTRIES` we use `[]`.
Thanks for checking!

**Comment by charliermarsh** [comment]
I feel like we _could_ catch this via the type annotation though?

**Comment by dhruvmanila** [comment]
Yeah, it should be doable

**Comment by gerrymanoim** [comment]
Just confirming that as of ruff 0.6.6 it still doesn't catch the constant via type annotations:

```
TEST = [1, 2, 3]

def test(l: list[int] = [1, 2, 3]):
    pass

def test2(l: list[int] = TEST):
    pass
```

Does https://github.com/astral-sh/ruff/pull/11122 need more work? Anything that we could help with?
